app/assets/javascripts/autocomplete.coffee

```diff
@@ -11,7 +11,7 @@ update_textfield = (list_element) ->
   $('.js-autocomplete-resource-id').val($(list_element).data('res-id'))
 
 ready = ->
-  $('.js-autocomplete-textfield').on('focus input', ->
+  $('.js-autocomplete-textfield').on('change keyup', ->
```

Comment:
@johndbritton Thanks for the explanation :smile: 

> The Javascript was triggering a search after every single character was typed

I thought this is controlled by the timeout period we set, not because the difference between  `keyup` and `input`. My understanding is that `keyup` would trigger more events than `input` when a user is typing (since `input` is triggered only when the `value` is changed).

> which led to exhausting the GitHub Search API rate limit of 30req/sec in some cases where a user was indecisive

A possible case that would exhaust the API is that a user keeps doing `focus` and `blur` (and this is a likely case).
In this case, `focus input` will trigger a lot of unnecessary requests compared to `change keyup`.

I was trying to follow up the design considerations here, it would be appreciated if you could give me some insights to help me clarify this.